extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -47,6 +47,9 @@ void additionalProviders(BuildProducer<AdditionalBeanBuildItem> additionalBean,
         additionalReaders
                 .produce(new MessageBodyReaderBuildItem(JacksonMessageBodyReader.class.getName(), Object.class.getName(),
                         Collections.singletonList(MediaType.APPLICATION_JSON)));
+        additionalReaders
+                .produce(new MessageBodyReaderBuildItem(JacksonMessageBodyReader.class.getName(), String.class.getName(),
```

Comment:
I don't see why we do this, given that we refuse to unserialise a `String` parameter in the JSON reader. `Object` above should just work and match `List<String>` via inheritance.

Reply:
This is needed, otherwise the order with which the reader is determined turns out to be wrong.

Reply:
OK, but why? Especially for a `List<String>` I don't think we should ever match with a `String` type, not for JSON.

It could make sense for the parameter convertors we support but those don't use readers.

Reply:
When I was looking I saw it was because we were looking up readers for `String`,  not `List`.
Now I am pretty sure that is a problem in itself, but I would prefer to keep this fix for now (as it can affect a lot of users) and deal with it more holistically if the need arises (i.e. we encounter more problems with reader selection).

Reply:
I'm not sure, I think looking up readers for `String` when we have `List<String>` is wrong and could lead to all sorts of troubles. We should find out why that is and not add weird workarounds that we'll surely forget to remove and then wonder why they are here ;)

Reply:
I've been burned by the readers too many times to have the will to fix it properly a few days before PTO :).

If we add the workaround, we will at least unblock a very common problem the code has now and thus allow us to see if there are any further problem. If there are, then I wholeheartedly agree to take a more extensive look at it.